gadopt/doc_descriptions.txt

```diff
@@ -0,0 +1,134 @@
+these variables are stored in Firedrake functions.
+
+- Developer guide
+
+Welcome to G-ADOPT's developer guide! In this guide, you will learn about the internal
+structure of the library and acquire the required knowledge to experiment with and
+contribute to the development of G-ADOPT.
+
+- Index
+
+G-ADOPT comprises several modules that provide a user-friendly API to easily set up
+physical approximations and mathematical solvers. The library currently offers
+dedicated solvers for the Stokes system (including a free surface), the energy
+equation, and the combined advection and reinitialisation of conservative level sets.
+Moreover, G-ADOPT incorporates specialised modules to tackle generic time-integration
+problems and optimised adjoint inversions. Additional helper tools for pre- and
+post-processing are provided and maintained.
+
+
+- Approximations
+
+This module provides classes that emulate physical approximations of fluid dynamics
+systems by exposing methods to calculate specific terms in the corresponding
+mathematical equations. Users instantiate the appropriate class by providing relevant
+parameters and pass the instance to other objects, such as solvers. Under the hood,
+G-ADOPT queries variables and methods from the approximation.
+
+- Diagnostics
+
+This module provides a class to simplify computing of diagnostics typically encountered
+in geodynamical simulations. Users instantiate the class by providing relevant
+parameters and call individual class methods to compute associated diagnostics.
+
+- Energy solver
+
+This module provides a fine-tuned solver class for the energy conservation equation.
+Users instantiate the `EnergySolver` class by providing relevant parameters and call
+the `solve` method to request a solver update.
+
+- Equations
+
+This module contains abstract classes to define the structure of mathematical terms and
+equations within the G-ADOPT library. Users should not interact with these classes;
+instead, please use the solvers provided in other modules.
+
+- Inverse
+
+This module provides classes to perform an adjoint inverse optimisation and checkpoint
+intermediate results. Users instantiate the `LinMoreOptimiser` class by providing
+relevant parameters and call the `run` method to perform the optimisation.
+
+- Level-set tools
+
+This module provides a set of classes and functions enabling multi-material
+capabilities. Users initialise materials by instantiating the `Material` class and
+define the physical properties of material interfaces using `field_interface`. They
+instantiate the `LevelSetSolver` class by providing relevant parameters and call the
+`solve` method to request a solver update. Finally, they may call the `entrainment`
+function to calculate material entrainment in the simulation.
+
+- Limiter
+
+This module provides a class that implements limiting of functions defined on
+discontinuous, linear function spaces. Users instantiate the class by providing
```

Comment:
Drop the work linear here - Steph has made progress on higher order limiting in a branch, albeit not merged yet.